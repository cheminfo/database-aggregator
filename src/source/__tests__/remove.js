'use strict';

const mongoSetup = require('../../../test/mongoSetup');
const { getCollection, clean } = require('../../../test/util');
const remove = require('../remove');

beforeAll(mongoSetup.connect);
afterAll(mongoSetup.disconnect);

const collection = getCollection('source_test');
const config = {
  driver: {
    getIds(config) {
      const data = ['test1', 'test3'];
      if (config.type === 'set') return new Set(data);
      return data;
    },
    getData() {
      // ignore
    }
  },
  collection: 'test'
};

const testData = {
  // eslint-disable-next-line camelcase
  meta_source_sequence: [
    {
      _id: 'test',
      seq: 3
    }
  ],
  // eslint-disable-next-line camelcase
  source_test: [
    {
      id: 'test1',
      commonID: 'test1',
      date: new Date(0),
      sequentialID: 1,
      data: {}
    },
    {
      id: 'test2',
      commonID: 'test2',
      date: new Date(0),
      sequentialID: 2,
      data: {}
    },
    {
      id: 'test3',
      commonID: 'test3',
      date: new Date(0),
      sequentialID: 3,
      data: {}
    }
  ]
};

describe('source remove', () => {
  beforeEach(async () => {
    await mongoSetup.dropDatabase();
    await mongoSetup.insertData(testData);
  });
  it('should ignore remove if threshold is too low', async () => {
    await remove(config);
    const data = await collection.find().toArray();
    expect(data[1].data).toEqual({});
    expect(clean(data)).toMatchSnapshot();
  });

  it('should remove if threshold is high enough', async () => {
    await remove(Object.assign(config, { removeThreshold: 0.9 }));
    const data = await collection.find().toArray();
    expect(data[1].data).toBeNull();
    delete data[1].date;
    expect(clean(data)).toMatchSnapshot();
  });

  it('should remove when driver returns a set', async () => {
    await remove(Object.assign(config, { type: 'set', removeThreshold: 0.9 }));
    const data = await collection.find().toArray();
    expect(data[1].data).toBeNull();
    delete data[1].date;
    expect(clean(data)).toMatchSnapshot();
  });
});
