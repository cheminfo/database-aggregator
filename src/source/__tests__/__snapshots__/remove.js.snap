// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source remove should ignore remove if threshold is too low 1`] = `
Array [
  Object {
    "_id": "test1",
    "commonID": "test1",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 1,
  },
  Object {
    "_id": "test2",
    "commonID": "test2",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 2,
  },
  Object {
    "_id": "test3",
    "commonID": "test3",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 3,
  },
]
`;

exports[`source remove should remove if threshold is high enough 1`] = `
Array [
  Object {
    "_id": "test1",
    "commonID": "test1",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 1,
  },
  Object {
    "_id": "test2",
    "commonID": "test2",
    "data": null,
    "sequentialID": 4,
  },
  Object {
    "_id": "test3",
    "commonID": "test3",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 3,
  },
]
`;

exports[`source remove should remove when driver returns a set 1`] = `
Array [
  Object {
    "_id": "test1",
    "commonID": "test1",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 1,
  },
  Object {
    "_id": "test2",
    "commonID": "test2",
    "data": null,
    "sequentialID": 4,
  },
  Object {
    "_id": "test3",
    "commonID": "test3",
    "data": Object {},
    "date": 1970-01-01T00:00:00.000Z,
    "sequentialID": 3,
  },
]
`;
